compiler/src/iree/compiler/Dialect/Util/Transforms/DumpModule.cpp

```diff
@@ -17,7 +16,13 @@
 
 namespace mlir::iree_compiler::IREE::Util {
 
-struct DumpModulePass : public DumpModuleBase<DumpModulePass> {
+#define GEN_PASS_DEF_DUMPMODULEPASS
+#include "iree/compiler/Dialect/Util/Transforms/Passes.h.inc"
+
+namespace {
+
+class DumpModulePass final : public impl::DumpModulePassBase<DumpModulePass> {
```

Comment:
nit: make the data member (i.e., `path) private because it is now a class.

EDIT: perhaps we can just move the `path` to the `options`, then we don't need the define the constructor in Passes.td. E.g., this is how it's done on CPU code: https://github.com/iree-org/iree/pull/18132/commits/fb8f053b830ad7d585decda286ebb2294e14d61f#diff-1e3b594a934d4a19e8f736d21de5e0a3f2e121b629a06a26247265c6d30089e4R16-R30

We still can have a wrapper for the pass. The default value should be gone in the declaration (i.e., `Passes.h`). So

```
std::unique_ptr<OperationPass<mlir::ModuleOp>>
createDumpModulePass(std::string path = "");
```

becomes

```
std::unique_ptr<OperationPass<mlir::ModuleOp>>
createDumpModulePass(std::string path);
```

Reply:
Thanks for your suggestions @hanhanW, I'll iterate on it :)